cryptographic_estimators/BIKEEstimator/bike_estimator.py

```diff
@@ -0,0 +1,104 @@
+        - ``truncate`` -- truncate rather than round the output (default: False)
+
+
+        EXAMPLES::
+
+        sage: from cryptographic_estimators.BIKEEstimator import BIKEEstimator
+        sage: A = BIKEEstimator(100, 10, 10)
+        sage: A.table()
+        +-------------+------------------+---------------+
+        |             |                  |    estimate   |
+        +-------------+------------------+------+--------+
+        | algorithm   |   attack_type    | time | memory |
+        +-------------+------------------+------+--------+
+        | SDKeyAttack |   key-recovery   | 20.2 |   15.5 |
+        | SDMsgAttack | message-recovery | 21.2 |   17.4 |
+        +-------------+------------------+------+--------+
+
+        sage: from cryptographic_estimators.BIKEEstimator import BIKEEstimator
+        sage: A = BIKEEstimator(150, 12, 11)
+        sage: A.table(show_all_parameters=1)
+        +-------------+------------------+--------------------------------------------------------------------+
+        |             |                  |                              estimate                              |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+        | algorithm   |   attack_type    | time | memory |                     parameters                     |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+        | SDKeyAttack |   key-recovery   | 21.8 |   16.6 | {'r': 5, 'p': 1, 'l': 9, 'SD-algorithm': 'Stern'}  |
+        | SDMsgAttack | message-recovery | 22.8 |   20.7 | {'r': 5, 'p': 2, 'l': 13, 'SD-algorithm': 'Stern'} |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+
+
+        TESTS::
+
+        sage: from cryptographic_estimators.BIKEEstimator import BIKEEstimator
+        sage: A = BIKEEstimator(3000, 50, 45)
+        sage: A.table(show_all_parameters=1) # long time
+        +-------------+------------------+--------------------------------------------------------------------+
+        |             |                  |                              estimate                              |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+        | algorithm   |   attack_type    | time | memory |                     parameters                     |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+        | SDKeyAttack |   key-recovery   | 57.0 |   33.7 | {'r': 9, 'p': 2, 'l': 41, 'SD-algorithm': 'Stern'} |
+        | SDMsgAttack | message-recovery | 58.3 |   33.7 | {'r': 9, 'p': 2, 'l': 41, 'SD-algorithm': 'Stern'} |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+
```

Comment:
maybe we should also add the real BIKE parameters as slow tests?

Reply:
we can, but it takes very long, didn't want to overload the CI even more